### DISCUSSION TITLE
trying out your basic board.

### DISCUSSION THREAD

**Discussion by pgpete** [bendecaf/xfm2-pcbs#11] [General]
Ive grabbed your basic board gerber files and got the basic board(s) made - (they are still in the mail)  I'm looking forward to testing them out. 

Your Advanced board looks good - except I dont have any experience soldering surface mount (yes :) - I realize this is an opportunity to learn) -  I  think SMC  might put off people that only have basic soldering skills and tools.  

cheers

Pgpete

**Comment by bendecaf** [comment]
That great. Let me know how it works out. I have made a few of these for both myself and a friend. They should work fine as long as you don't use the uart breakout pins. The uart pins probably need a pull-up resistor, a buffer or something like that to work better. 

About the advanced board: I still have to build and test this myself, possibly next week. About the SMD design, I know that the diy barrier is higher. However, it's part of something larger that I mess around with, so I thought I might just as well do it all SMD based. I'm not sure yet where it all leads to.

Anyway: good luck!

**Reply by pgpete** [reply]
Hi Bzeiss -  boards arrived and i put one together - no problems.  the board works for both XFM2 and VA1 -  I'm now struggling to  maintain reliable USB communications  so I can  do patch change and edit patches.  Now that Ive got this far hardware wise - I'm thinking of  getting your advance board made.  and maybe  bread-board up some  tweaking knobs.

have you built up your advanced board yet ?

**Reply by bendecaf** [reply]
Nice! I don't think stability of the USB connection will be any different with the extended board though. Are you referring to instabilities with the Excel sheet and the USB connection? I think there is a comment about the on the xfm2/xva1 blog. The XVA 1.01 has a more recent and more stable version of the Excel sheet than XFM2.

About the extended board: yes, I have tested it, but it needed a new revision in order to work properly. There is a problem with the A100 connector though. It's currently on the way to me. If you intend to bread board knobs with an arduino or similar, I suggest you wait for a working new revision for the reason alone that I have added logic level converters and pull-ups for the uart pins which make working with the uart pins with external arduinos/microcontrollers much easier and more stable.

**Reply by bendecaf** [reply]
So unfortunately, there are still a few problems with the new board. I'll have to do some analysis and make a new revision. So it'll still take a few weeks until it's ready.

**Reply by pgpete** [reply]
yep - I'm keen to make the debugged extended board -  also very excited about the front panel.

**Reply by bendecaf** [reply]
That's great. I have the new pcbs already on my desk, but I didn't have a chance to solder one together. After all, it takes a while to solder it and I have switched working on a control board UI for RASCSI. I think my other project will be finished within a few weeks. I'll continue to work on these boards after that.

**Reply by kurtblah** [reply]
Hey, -any news on revised boards? The extended board is a great idea and looks awesome.

**Comment by bendecaf** [comment]
Sorry not yet. Maybe I'll have a chance to solder one next weekend.

**Reply by bendecaf** [reply]
@pgpete  @kurtblah 
I have soldered and successfully tested the following things:

* programming the flash memory and the fpga through the micro usb connector
* midi in, midi out, midi thru (midi out through midi-ox and the ctrl alpha panel)
* spdif
* analog out
* leds
* A100 connector
* barrel-jack power connector
* A100 + barrel-jack power connector connected at the same time
* UART through the usb-c connector through teraterm

There are two things that I have noticed so far:
- After switching to Windows 11 on my development machine, the excel vba files do not work any more for some reason (at least I think that is the reason why). Due to that, it's not easy for me to actually play around with the parameters and thus test the CV1/CV2 connectors as they require some fiddling with the default presets. I have to find a solution for that.
- My control panel prototype doesn't work at the moment for a reason I still have to debug. As a result, I haven't properly tested the uart logic level converters yet.
- I wanted to put a shrouded connector for the control board connector on this time so that it can't be connected the wrong way. I have noticed that there is too little space for that on the 1.34 version. So I have to move the connector a bit.

In other words: while the last revision looks promising and while I am pretty sure that this one probably works without bigger issues, there are still some things to work out. My current plan is to revise the board revision to a 1.4 revision with the pin breakout connector moved a little, update to kicad 6 and create assembly files for jlcpcb as I really don't want to spend any more time than necessary soldering this again to the full extend. So I at the very least want capacitors, resistors and leds placed already, possibly other ics and diodes as well. 

While this is being produced, I'll try to find a solution for the Excel problem. It's likely I'll just end up upgrading that alpha ctrl panel with the missing CV1/CV2 features and what else is missing.

**Reply by pgpete** [reply]
well done!

**Reply by bendecaf** [reply]
The control voltage opamp works as well with the hacked ctrlr panel.

**Reply by kurtblah** [reply]
nice

**Reply by bendecaf** [reply]
@kurtblah @pgpete So I have received the boards with the preassembled parts. They seem to work mostly fine in general, however, it seems I have unfortunately some more to work to do on the power circuit. When the board is driven through the external power connectors (A100 or the power jack), the power goes through LM1117 power regulators which are rated with a max of 1A (for the chinese versions I use). It seems I somehow underestimated the draw of the fpga when I attach my front panel board with microcontroller and lcd display as well. Once I do that, the voltage regulator starts to make funny sounds. Without it, it just gets warm. It seems the texas instruments regulators I have used previously are more tolerant at an actually lower rating of 800mA. After checking the datasheets, the power circuit is probably underspecified even for the fpga. So currently, I don't recommend using this circuit (except if you plan to power it through the cmod a7 via micro usb and probably also without additional power draw from the breakout pins).

Edit: ok, after actually measuring, it's maxing out at 390mA. There is something else going on. Possibly the Chinese voltage regulators aren't good.

**Reply by bendecaf** [reply]
So I have added a few details to the wiki page. I think it's ready to be used, I have added a few important notes about the thermal (and current) constraints of the device though.

**Reply by conversy** [reply]
Hi,

I have ordered a few of your basic boards, thank you for sharing the sources!

I was trying out the bitstreams over at https://www.futur3soundz.com/da-blog/xfm2-testing-the-build . `Tests_blinky.bit`seems to work, but not `Tests_memory.bit` nor `Tests_audio.bit`. Would you please be kind enough to confirm that they work with your boards?

Since the adafruit audio board is not available anymore, I bought a clone on aliexpress, maybe the audio problem is related to a faulty audio board...

Best!

**Comment by bendecaf** [comment]
Hi @conversy, I won't be able to test these until Mid January as I don't have access to my boards and tools until then. However, the basic board is essentially the original schematic on a pcb. So I'm pretty sure that they are supposed to work. You can maybe post a picture of you build. Yes, AliExpress clones are hit and miss sometimes. I have always used original adafruit boards from mouser. Also, check the pin orientation of the memory and audio modules.

**Reply by conversy** [reply]
Thanks for the tips. I double-checked the orientation of the chips. Still, the first time I plugged the CMOD board, I did it in a reverse position :-/ I hope I did not fried the audio card or the EEPROM... I'll check with my arduino that each chip is still working.
Mouser does not have adafruit board anymore, unfortunately...

**Reply by conversy** [reply]
Hi,

I still have no luck with my setup. This time I used a genuine adafruit breakout board, to no avail.
- `test_blinky.bit` from [xfm2-testing-the-build](https://www.futur3soundz.com/da-blog/xfm2-testing-the-build) is ok ✔️
- `tests_memory.bit` is not ok, the led is purple, I guess it's red+blue 'Unable to write memory' (this is the result with a bare CMOD board)'
- `tests_audio.bit` spits out nothing
- `XVA1_CMOD_A7.bin` receives midi (led blinking), but no audio is produced.

I thought that the unit should be working even with no tunings nor EEPROM, am I right?
Does it smell like bad solder connections?

**Reply by AlteredWolf** [reply]
> Hi,
> 
> I still have no luck with my setup. This time I used a genuine adafruit breakout board, to no avail.
> 
> * `test_blinky.bit` from [xfm2-testing-the-build](https://www.futur3soundz.com/da-blog/xfm2-testing-the-build) is ok ✔️
> * `tests_memory.bit` is not ok, the led is purple, I guess it's red+blue 'Unable to write memory' (this is the result with a bare CMOD board)'
> * `tests_audio.bit` spits out nothing
> * `XVA1_CMOD_A7.bin` receives midi (led blinking), but no audio is produced.
> 
> I thought that the unit should be working even with no tunings nor EEPROM, am I right? Does it smell like bad solder connections?

Just a note: I have, in the past, received bad audio boards. My first XFM2 build didn't work until I swapped-out the defective audio board for a new one. Thankfully, I had ordered several of them since they were so cheap.

 ---I should point out that I was using my own pcb, not the bzeiss one.

**Reply by conversy** [reply]
thanks @AlteredWolf I used both a clone and a genuine board, but no sound with both :-/. Problem is, I don't know how to debug this. Maybe with a scope, time to get some new skills...

**Reply by bendecaf** [reply]
If you have a multimeter, you could test the pcb traces for continuity. As the basic board is essentially just the original schematic (https://www.futur3soundz.com/xfm2) as a pcb, you can test the connections exactly as depicted in the schematic. If those traces test correctly, it's likely not the pcb. The next step would then be to test your breakout board standalone with a micro controller or a raspberry pi, for example, by following the adafruit tutorials: https://learn.adafruit.com/adafruit-i2s-stereo-decoder-uda1334a. Lastly, also check your soldering, maybe post images for your soldering here for us to check it. It happened to me more than once with through-hole soldering that I simply missed to solder a pin. If you missed or skipped one of the used ones, it likely won't work.

**Reply by conversy** [reply]
![IMG_0229](https://user-images.githubusercontent.com/3647733/212971235-85a8e7ab-8354-4938-b578-92e538f7aff1.jpg)
![IMG_0228](https://user-images.githubusercontent.com/3647733/212971271-c4fd2f41-fa75-401a-b03c-0cdbe008b3bc.jpg)

**Reply by conversy** [reply]
I've tested:
DGND-VIN: 3.3v
DGND-SF1: 3.3v
DGND-MUTE: 0v
DGND-SF0: 3.3v
DGND-SCK: 1.6v
DGND-SDA: 0v
DGND-WS: 1.6v

Thanks for helping!

EDIT: what boggles me is that I am able see the led blinking when sending midi, but no led blinking for audio production. I have tested it without flashing the tunings or the eeprom with the presets.

EDIT2: I tried to upload the tunings by first launching `bitloader.exe` then `XLoad "t XVA1_Tunings.bin"`, I can see "Erasing Flash..." in green, but nothing happens next, it's stuck.

**Comment by bendecaf** [comment]
These are the traces for continuity testing. Maybe these help. The lines except VCC and GND are data lines that will go high and low. So the voltages don't matter a lot on everything except VCC and GND. You should test if the VCC pins below have 3.3V and that GND is 0V. You won't be able to see much on the other lines if you don't have a logic analyzer or an oscilloscope.

VCC:
![grafik](https://user-images.githubusercontent.com/884834/212998611-26a1465b-25fd-4fc1-832d-156cd32e4f31.png)
MUTE:
![grafik](https://user-images.githubusercontent.com/884834/212998667-e364d9f4-5c46-47fd-8f48-36660bf3ff09.png)
SCK:
![grafik](https://user-images.githubusercontent.com/884834/212998726-0d926182-3cb4-4d33-aa0c-8ea65f04934a.png)
SDA:
![grafik](https://user-images.githubusercontent.com/884834/212998770-a406e54b-3dc7-4816-9cdb-662e670008aa.png)
WS:
![grafik](https://user-images.githubusercontent.com/884834/212998812-75707a71-05fc-4cad-ac5a-828fc647ca3c.png)
GND:
![grafik](https://user-images.githubusercontent.com/884834/212998853-f3691a7e-7e53-4341-9c82-1019180975fa.png)

You might want to check your solder joints in a few places. I can't tell from the pictures whether these are reflections or incompletely soldered joints.
![grafik](https://user-images.githubusercontent.com/884834/212999896-4b8f2dca-002e-488d-9f9b-934a882302dc.png)
![grafik](https://user-images.githubusercontent.com/884834/213000115-826a94b3-75b1-461c-98aa-8a5550cecb77.png)

**Comment by bendecaf** [comment]
About that hanging bitloader: it's behaving a bit stubborn sometimes. I have tested various versions of the loader and found that the tooling as bundled in the releases is not always the most reliable. There are also some blog posts on rene's web page that suggest how to handle certain behaviors. Unfortunately, my directory where I have assembled the versions seems to have vanished with various system changes I have made last year. I can only suggest to download a couple of xva releases and try different versions of it.

**Reply by conversy** [reply]
Thank you very much. May I ask you something again? Is the audio led supposed to blink when receiving midi events, meaning that it produces audio, even if there is no i2s board or eeprom? Is it the case with your build? The fact that mine isn't even blinking though the midi led is, is very suspect...

(as for the joint I *think* these are reflections, but I'll double check)

**Reply by bendecaf** [reply]
I currently don't have a basic board at hand. I'll have to check if I have a board left somewhere in some drawer. I may get back to you on this on the weekend. Unfortunately, I won't have time to do searching and experiments before that.

**Reply by conversy** [reply]
Thank you! I guess it would be the same with your more complex board: if you only keep the CMOD and the midi input, does the audio led on the CMOD blink when receiving midi input? Thank you again!

**Comment by bendecaf** [comment]
So I have found my basic board, but it seems I have soldered the DAC directly onto the board. I'll probably resolder and socket everything on the weekend. I don't have a spare extended board either. I have to build another one when I find the time, but this ones takes a lot more time even with my preassembled boards :-)

So one piece of good news though. I have downloaded the XVA1 package once more to flash another brand new CMOD A7 that I had fresh from Mouser (for a few months actually) and got stuck with the Tunings as well. I think, I remember now what the issue was and just confirmed it by trying to reflash the cmod. Once your done with flashing the XVA1_CMOD_A7.bin as the synth firmware, you have to disconnect the cmod from the usb port so that it doesn't have power. Then you reconnect, load the tunings file without loading the bitloader first. Then it should work. The one where it gets a little tricky is when you want to load the XFM2 test version with sample support.

**Comment by bendecaf** [comment]
Found my old instruction notes:
01) Open cmd as administrator
02) cd bitloader
03) bitloader.exe
04) cd..
05) xload -img XVA1_CMOD_A7.bin
06) unplug the usb cable and plug in the usb cable again. Wait a few seconds.
07) xload "t XVA1_Tunings.bin"
08) xload "put_bank XVA1_Factory.bank"
09) xload "* 0"

**Reply by conversy** [reply]
Great, thank you, I'll try when I'm in front of a windows computer again!

**Comment by bendecaf** [comment]
So I have managed to test my new cmod on an extended board with sockets and if I put data through midi in, the cmod blinks with with both the blue led (midi in signal) and green led (audio signal) also when the uda1334 is not plugged in.

**Reply by conversy** [reply]
great, that was the test I expected. As soon as I'm able to apply your instructions, I'll know if my cmod is the culprit. Thanks again!

**Reply by conversy** [reply]
So, I tried your procedure, and it works until I set the midi channel: a led is shortly blinking on the board, but `XLoad.exe` does not exit.
Anyway, still no luck with the sound, no audio is produced. I have a new CMOD board, but I get the same result. The midi reception works, but not the audio generation.
In addition, the audio test does not work, neither the test with memory.
I'm really wondering what's going on, at this point I do not know what to do.

**Reply by conversy** [reply]
... and for some reason, XFM2 works! So I *must* do something wrong with XVA1, or it's picky about something?!

**Reply by bendecaf** [reply]
Well done, one step further! I don't know, XVA1 does work fine for me. You have loaded both the tunings and the bank, right? I can only imagine it's something rather simple. Maybe try to change the XVA1 sound/program via a terminal application like teraterm on Windows or the Excel sheet if this works for you. I've had situations where parameters of the first program were borked for some reason.

**Reply by conversy** [reply]
Yes I loaded both the tunings and the bank. XLoad does not return when changing the midi channel though. I'll try without setting it, and by following your advice on the terminal and Excel.
A lesson learnt though: the test for audio does not work, but XFM2 works, so I guess the tests are not valid anymore.
